### PR TITLE
CLN/TST: Clean groupby.test_allowlist

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

The allowlist has been removed, so this test file should be cleaned up. Renamed `test_api_consistency` to `test_api` so we can group some of these tests. The tests that remain in test_allowlist are likely duplicates of other tests and can probably be removed. Will look to do so in a followup.

### CODE REVIEW

File: pandas/tests/groupby/test_allowlist.py [hunk 1]
```diff
@@ -100,131 +95,6 @@ def test_regression_allowlist_methods(raw_frame, op, axis, skipna, sort):
         tm.assert_frame_equal(result, expected)
 
 
-def test_groupby_blocklist(df_letters):
```
Comment (rhshadrach) [inline pandas/tests/groupby/test_allowlist.py:103]: Removing this test - there we no longer try to get to ops through `__getattr__`.

File: pandas/tests/groupby/test_allowlist.py [hunk 2]
```diff
@@ -100,131 +95,6 @@ def test_regression_allowlist_methods(raw_frame, op, axis, skipna, sort):
...
-        "expanding",
-        "pipe",
-        "sample",
-        "ewm",
-        "value_counts",
-    }
-    assert results == expected
-
-
-def test_groupby_function_rename(mframe):
```
Comment (rhshadrach) [inline pandas/tests/groupby/test_allowlist.py:221]: Removing this test, we no longer dynamically add on groupby methods.

**Review by topper-123 — APPROVED** [review]
LGTM. There's not a lot left in test_allowlist. Can the remaining tests be moved to other locations?

**Comment by topper-123** [comment]
thanks @rhshadrach

**Pull request merged**